xonsh/completer.py

```diff
@@ -311,20 +340,11 @@ def bash_complete(self, prefix, line, begidx, endidx):
         rtn = {s + space if s[-1:].isalnum() else s for s in out.splitlines()}
         return rtn
 
-    def _source_completions(self):
-        srcs = []
-        for f in builtins.__xonsh_env__.get('BASH_COMPLETIONS'):
-            if os.path.isfile(f):
-                # We need to "Unixify" Windows paths for Bash to understand
-                if ON_WINDOWS:
-                    f = RE_WIN_DRIVE.sub(lambda m: '/{0}/'.format(m.group(1).lower()), f).replace('\\', '/')
-                srcs.append('source ' + f)
-        return srcs
-
     def _load_bash_complete_funcs(self):
+        """Attempt to load any Bash completion functions we can find."""
         self.bash_complete_funcs = bcf = {}
-        inp = self._source_completions()
-        if len(inp) == 0:
+        inp = _source_bash_completions()
+        if not inp:  # no completion files could be found
```

Comment:
> Please return this to the original positive conditional. 

No. The original "positive conditional" is both needlessly verbose and antithetical to the Python maxim that there should be one obvious way to do something. In this case we're asking whether or not `inp` contains anything of interest. It doesn't matter if it is a sequence of length zero or None.